src/goto-programs/json_goto_trace.cpp

```diff
@@ -114,6 +111,13 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
+          if(step.full_lhs_value.is_not_nil())
+            full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
```

Comment:
I added the assertions and run the tests that are normally run on CBMC and test-gen adding the `--json-ui` and no assertion failure was triggered